src/extensions/auth/permit/index.ts

```diff
@@ -0,0 +1,227 @@
+import { AminoSigner, StdSignDoc } from "../../../wallet_amino";
+import { AminoWallet, serializeStdSignDoc } from "../../../wallet_amino";
+import { bech32 } from "bech32";
+import { base64PubkeyToAddress } from "../../../index";
+import * as secp256k1 from "@noble/secp256k1";
+import { fromBase64 } from "@cosmjs/encoding";
+import { sha256 } from "@noble/hashes/sha256";
+
+export class PermitError extends Error {
+  readonly type = "PermitError";
+  constructor(message: string) {
+    super(message);
+    this.name = "PermitError";
+  }
+}
+
+export class ContractNotInPermit extends PermitError {
+  contract: string;
+  allowed_contracts: string[];
+
+  constructor(contract: string, allowed_contracts: string[]) {
+    super("No contract in : " + contract);
+    this.name = "ContractNotInPermit";
+    this.contract = contract;
+    this.allowed_contracts = allowed_contracts;
+  }
+}
+
+export class SignatureInvalid extends PermitError {
+  signature: string;
+  key: string;
+
+  constructor(signature: string, key: string) {
+    super("Signature invalid in permit");
+    this.name = "SignatureInvalid";
+    this.key = key;
+    this.signature = signature;
+  }
+}
+
+export class SignerIsNotAddress extends PermitError {
+  publicKey: PubKey;
+  address: string;
+
+  constructor(publicKey: PubKey, address: string) {
+    super("Address is not signer");
+    this.name = "SignerIsNotAddress";
+    this.address = address;
+    this.publicKey = publicKey;
+  }
+}
+
+export class PermissionNotInPermit extends PermitError {
+  permission: Permission[];
+  permissionsInContract: Permission[];
+
+  constructor(permission: Permission[], permissionsInContract: Permission[]) {
+    super("Address is not signer");
```

Comment:
Is this supposed to be here? 🤔

Reply:
Yeah, just a different error message. Thanks for pointing it out, I'll fix shortly!